enclave_build/src/pull.rs

```diff
@@ -0,0 +1,384 @@
+    #[cfg(target_arch = "aarch64")]
+    const SAMPLE_IMAGE: &str =
+        "667861386598.dkr.ecr.us-east-1.amazonaws.com/enclaves-samples:vsock-sample-server-aarch64";
+    #[cfg(target_arch = "x86_64")]
+    const IMAGE_HASH: &str =
+        "sha256:d63f69841675f849534670141c0f7106f1f2faa5d4bf02ea8f144bd3ca292f80";
+    #[cfg(target_arch = "aarch64")]
+    const IMAGE_HASH: &str =
+        "sha256:0af5126ba4d02a5b64c831bce95a8aa98e20b0ed9ea60174dae19000de813a6c";
+
+    const IMAGE_REG: &str = "667861386598.dkr.ecr.us-east-1.amazonaws.com";
+    const TEST_USER: &str = "test_username";
+    const TEST_PASS: &str = "test_password";
+
+    fn generate_valid_credential_json() -> Value {
+        let credentials = TEST_USER.to_string() + ":" + TEST_PASS;
+        let encoded_token = base64::encode(credentials);
+
+        json!({
+            "auths": {
+                IMAGE_REG: {
+                    "auth": encoded_token
+                }
+            }
+        })
+    }
+
+    fn generate_credential_json_missing_auth() -> Value {
+        json!({
+            "auths": {
+                IMAGE_REG: {},
+                "reg1": {
+                    "auth": "token1"
+                },
+                "reg2":  {
+                    "auth": "token2"
+                },
+            }
+        })
+    }
+
+    fn generate_credential_json_invalid_encoding() -> Value {
+        json!({
+            "auths": {
+                IMAGE_REG: {
+                    "auth": "not_base64"
+                },
+            }
+        })
+    }
+
+    fn generate_credential_json_invalid_reg() -> Value {
+        let credentials = TEST_USER.to_string() + ":" + TEST_PASS;
+        let encoded_token = base64::encode(credentials);
+
+        json!({
+            "auths": {
+                "https://index.docker.io/v1/": {
+                    "auth": encoded_token
+                }
+            }
+        })
+    }
+
+    fn create_auth_file(content: &Value) -> TempFile {
+        let auth_file = TempFile::new().unwrap();
+        let json_bytes = serde_json::to_vec(content).unwrap();
+        auth_file.as_file().write_all(&json_bytes[..]).unwrap();
+
+        auth_file
+    }
+
+    /// Extract username from the base64 encoded credential token. Password can't be known
+    #[test]
+    fn test_parsing_credentials() {
+        let auth_content = generate_valid_credential_json();
+        let tmp_file = create_auth_file(&auth_content);
+        let config_path = tmp_file.as_path();
+        std::env::set_var("DOCKER_CONFIG", config_path);
+
+        let uri = SAMPLE_IMAGE;
+
+        let credentials = parse_credentials(uri).unwrap();
+
+        if let RegistryAuth::Basic(username, password) = credentials {
+            assert_eq!(username, TEST_USER);
+            assert_eq!(password, TEST_PASS);
+        } else {
+            unreachable!();
+        }
+    }
+
+    /// Test different credential parsing errors
+    fn test_parsing_failure(content: &Value) {
```

Comment:
nit: would be nice to rename helper functions so that they don't start with `test_`. It is confusing as you would expect this is a proper test, not something called by the actual tests.